js/chips.js

```diff
@@ -0,0 +1,36 @@
+// some mock functions
+
+function selectChip() {
+	// things that happen when a user clicks-on a chip on the board:
+	// - get the ID of the selected chip
+	// - highlight the chip in the UI so the user can see it's selected
+	// - get the ID of the current space/square/box that the selected chip is in
+	// - check the surrounding spaces for available valid moves
+	// - call the validateMoves function, pass the selected chip and the current square as arguments. 
+	//   might also need to pass-in the player or player's chip color
+	// validateMoves(selectedChip, currentSquare); 
+}
+
+function validateMoves(selectedChip, currentSquare) {
+	// things to check to find out what the available moves are for the selected chip:
+	// - get the array of nextValidSquares from the square's markup (or encapsulate that data into another object somewhere?)
+	// - check each of the nextValidQuares for empty squares or enemy chips/friendly chips using the squareContains function
+	// - create an array of available empty valid squares that the chip can move to
+	// - create an array of valid squares that contain enemy chips
+	// - when a valid square contains an enemy chip, check for available capture
+	// - if a capture is available, create an array of valid squares to *jump-to* that would *capture* an enemy chip
+}
+
+function squareContains(selectedChip) {
```

Comment:
Not sure if I understand this part. Shouldn't the function squareContains receive a parameter of square(s)? Such as squareContains(validateMoves) _parameter is an array, check for each in array_ or squareContains(validSquare) _checks for contents of a valid square.

The chip doesn't contain information about the next valid moves yet, if I remember correctly. Or will the selectedChip contain that info in the future, like stored in an temporary object?

Reply:
In this case my thought process was that there would be an array of `nextValidSquares` and that at some point we'd loop through the array and run a `squareContains()` function for each of those squares in the array. You're right that the function argument would take the ID of a square and not the `selectedChip`, I probably was just on a copy/paste run when I put that argument here 😬